### ISSUE TITLE
astropy tables containing unicode characters don't print in 2.7 (sometimes)

### ISSUE THREAD

**Issue by keflavich** [astropy/astropy#4944]
Astropy tables containing unicode sometimes fail to print with UnicodeDecodeErrors.  An example is in this issue: https://github.com/astropy/astroquery/issues/689

However, I can't reproduce this with a simple MWE:

```
# in py3:
T = Table()
T.add_column(table.Column(data=['blah','bläh'], name='ok'))
T.write('blah2.xml', format='votable') # votables seem to be the only format that support unicode?
```

then:
```
# in py2:
In [1]: from astropy.table import Table

In [2]: T = Table.read('blah2.xml')

In [3]: T
Out[3]:
<Table masked=True length=2>
   ok
unicode4
--------
    blah
    bläh
```

It looks like maybe there is a type mismatch between the dtype and the data in the astroquery column.  I haven't tracked this down fully yet, but I suspect it is a bug in the votable parser?  Or worse, yet another incorrectly specified votable... but might it be possible to automatically detect unicode in a votable column to prevent this sort of issue?

**Comment by pllim** [comment]
Looking at the traceback in your other issue, it looks very similar to something I encountered in Cone Search two months ago. I fixed that in Cone Search via #4757. 

My hunch is that this block in `table/pprint.py` needs modification:
```python
if six.PY3:
    def default_format_func(format_, val):
        if isinstance(val, bytes):
            return val.decode('utf-8')
        else:
            return str(val)
    _format_funcs = {None: default_format_func}
elif six.PY2:
    _format_funcs = {None: lambda format_, val: text_type(val)}
```

1. That block is not flexible enough to support Python 4 (not particularly related to your problem here).
2. There is `from __future__ import unicode_literals` on top of the file as well. So, maybe just use Python 3 logic regardless?
